### DISCUSSION TITLE
Onnx VN Comparison

### DISCUSSION THREAD

**Discussion by MarvNC** [Blinue/Magpie#846] [General]
I downloaded all the models available in the [linked repo here](https://github.com/hooke007/dotfiles/releases) and used ChaiNNer to test them all on a couple screenshots. Here's a [slow.pics for ハミダシクリエイティブ](https://slow.pics/c/yFfwtGsy), [slow.pics for オトメ＊ドメイン](https://slow.pics/c/U3UPM9vZ).

There were many good ones but I think I liked these the most. Didn't take that much time to pixel peep every one tho, and of course the results will depend a lot depending on the art/resolution/text rendering/etc of the source image.

- nagadomi_upconv_7_anime_style_art_rgb_x2_fp32_opset15
- tsurumeso_upresnet10_x2_fp32_opset15
- Phhofm_HFA2kAVCCompact_x2_fp32_opset14
- CuNNy

|                                                                                                           ハミダシクリエイティブ                                                                                                           |                                                                                          オトメ＊ドメイン                                                                                           |
| :----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: | :-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
| ![hamidashi_ハミダシクリエイティブ_-_Ver1 1_『本日より新ガチャ開催です』_2024-03-10_15-27-27-nagadomi_upconv_7_anime_style_art_rgb_x2_fp32_opset15](https://github.com/Blinue/Magpie/assets/17340496/b5da5602-02b2-416d-9d89-372a577d1b7c) | ![オトメドメイン_オトメ＊ドメイン_2024-03-10_13-13-02-nagadomi_upconv_7_anime_style_art_rgb_x2_fp32_opset15](https://github.com/Blinue/Magpie/assets/17340496/dab574ce-f875-4e8e-b86f-17752d0161be) |
|                                                                                           nagadomi_upconv_7_anime_style_art_rgb_x2_fp32_opset15                                                                                            |                                                                        nagadomi_upconv_7_anime_style_art_rgb_x2_fp32_opset15                                                                        |
|        ![hamidashi_ハミダシクリエイティブ_-_Ver1 1_『本日より新ガチャ開催です』_2024-03-10_15-27-27-Phhofm_HFA2kAVCCompact_x2_fp32_opset14](https://github.com/Blinue/Magpie/assets/17340496/43637864-4792-4af7-8384-71f2cf7ef0fc)         |        ![オトメドメイン_オトメ＊ドメイン_2024-03-10_13-13-02-Phhofm_HFA2kAVCCompact_x2_fp32_opset14](https://github.com/Blinue/Magpie/assets/17340496/feea9baa-c02d-4aa4-9749-ee37b125affc)         |
|                                                                                                   Phhofm_HFA2kAVCCompact_x2_fp32_opset14                                                                                                   |                                                                               Phhofm_HFA2kAVCCompact_x2_fp32_opset14                                                                                |
|         ![hamidashi_ハミダシクリエイティブ_-_Ver1 1_『本日より新ガチャ開催です』_2024-03-10_15-27-27-tsurumeso_upresnet10_x2_fp32_opset15](https://github.com/Blinue/Magpie/assets/17340496/a4effed0-7bab-4144-8789-3d9371bb39e3)          |         ![オトメドメイン_オトメ＊ドメイン_2024-03-10_13-13-02-tsurumeso_upresnet10_x2_fp32_opset15](https://github.com/Blinue/Magpie/assets/17340496/24e4f324-93c3-4b12-a41a-904157f266ac)          |
|                                                                                                    tsurumeso_upresnet10_x2_fp32_opset15                                                                                                    |                                                                                tsurumeso_upresnet10_x2_fp32_opset15                                                                                 |
|                                                                     ![CuNNy_hamikuri](https://github.com/user-attachments/assets/5c890f06-2021-41c9-9b00-f947b3bc69be)                                                                     |                                                  ![CuNNy_otodome](https://github.com/user-attachments/assets/bdd21a4c-bba2-4316-a90b-343863d3799a)                                                  |
|                                                                                                          CuNNy new 8x32 (2024-08)                                                                                                          |                                                                                      CuNNy new 8x32 (2024-08)                                                                                       |

Hamikuri title screen: https://slow.pics/c/MPVFnV0b

Also a couple comparisons for Tsukihime 640x480 upscaling:

https://slow.pics/c/M0fUBN9n
https://slow.pics/c/rE6x4uJN

**Comment by kato-megumi** [comment]
Anime4k's result

Anime4K_Upscale_UL
![Upscale_UL-otome](https://github.com/Blinue/Magpie/assets/29451351/bd640ad2-9181-4ff6-b6f4-9f94c77a393d)
![Upscale_UL-hamidashi](https://github.com/Blinue/Magpie/assets/29451351/c90e4ae7-8766-46fa-ba5d-46bdbda45fc2)
Anime4K_Upscale_UL Anime4K_Restore_Soft_UL
![Upscale_UL-Restore_Soft_UL-otome](https://github.com/Blinue/Magpie/assets/29451351/8170d918-abc0-47a7-a723-9722648ad317)
![Upscale_UL-Restore_Soft_UL-hamidashi](https://github.com/Blinue/Magpie/assets/29451351/9d25fe94-86b1-438e-acd8-5b23f1acf223)

Look remarkably close to waifu2x

**Comment by kato-megumi** [comment]
My attempt at dealing with ringing. Quite bad but still better than existing model.
https://slow.pics/c/C5wb8RHv

**Comment by rampaa** [comment]
Any possibility for 640x480 (e.g., [Tsukihime](https://vndb.org/r339)) to 4k comparison?

**Reply by MarvNC** [reply]
Can you send an image for comparison?

**Reply by rampaa** [reply]
Sure: https://i.imgur.com/sJh8Xjv.png, https://i.imgur.com/xEemAZ3.png

**Reply by MarvNC** [reply]
https://slow.pics/c/M0fUBN9n
https://slow.pics/c/rE6x4uJN
